xdsl/tools/tblgen_to_py.py

```diff
@@ -474,7 +474,7 @@ def generate_op(self, tblgen_op: TblgenOp, dialect_name: str):
                     fields[name] = "var_region_def()"
                 case (True, True):
                     fields[name] = 'var_region_def("single_block")'
-                case _:
+                case _:  # pyright: ignore[reportUnnecessaryComparison]
```

Comment:
Wait, should we just remove the default case then? We had this just because pyright wasn't happy, but now it is

Reply:
Yep I'll remove the default case with the new pyright, the old one still wants it. I guess I could remove it now and ignore the pyright error on the match case directly, I'll do that quickly.